cpp/test/compat/core/compat.cmake

```diff
@@ -20,3 +20,20 @@ if (CSHARP_COMPAT)
         run (${CSHARP_COMPAT} ${TEST} compat.Compat.json compat.${TEST}.gbc.dat)
     endif()
 endif()
+
+if (JAVA_COMPAT)
+    if (WIN32)
+        set (PATHSEP "\\\;")
+    else()
+        set (PATHSEP ":")
+    endif()
+
```

Comment:
Consider making this error out if `JAVE_CORE` isn't set, as `JAVA_COMPAT` implies that `JAVA_CORE` is also set.